multiqc/modules/fastqc/assets/js/multiqc_fastqc.js

```diff
@@ -172,10 +172,33 @@ $(function () {
             total += 1;
             v[status] += 1;
         });
+        var pass_percent = (v['pass']/total)*100;
+        var warn_percent = (v['warn']/total)*100;
+        var fail_percent = (v['fail']/total)*100;
+        var pass_number_of_digits = v['pass'].toString().length;
+        var warn_number_of_digits = v['warn'].toString().length;
+        var fail_number_of_digits = v['fail'].toString().length;
+        var max_percent = Math.max(pass_percent, warn_percent, fail_percent);
+        var pass, warn, fail; // variable used to save the width in a progress-bar
+        // Each digit needs around 8px of space
+        // The progress bar has a total width of 100px, so spaces allocated for pass, warn or fail will be calculated by their percentage * 100px
+        if (max_percent == pass_percent) {
+            warn = warn_percent != 0 && warn_percent < warn_number_of_digits * 8 ? warn_number_of_digits * 8 : warn_percent;
```

Comment:
Can this calculation be refactored into a function? It's used frequently, and is hard to understand in the one line.

Please also rename `pass`, `warn`, `fail` as declared on line 182, to be clearer about their purposes - right now lines 199-201 use both `fail` and `v['fail']` which look similar but have very different values and purposes.